compiler/x/codegen/X86OpcodeTable.hpp

```diff
@@ -0,0 +1,140 @@
+/*******************************************************************************
+ * Copyright IBM Corp. and others 2023
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at http://eclipse.org/legal/epl-2.0
+ * or the Apache License, Version 2.0 which accompanies this distribution
+ * and is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception [1] and GNU General Public
+ * License, version 2 with the OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] https://openjdk.org/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#ifndef OMR_X86OPCODETABLE_HPP
+#define OMR_X86OPCODETABLE_HPP
+
+#include <cstdint>
+#include "codegen/InstOpCode.hpp"
+
+enum ArithmeticOps : uint32_t
+   {
+   ArithmeticInvalid,
+   BinaryArithmeticAdd,
+   BinaryArithmeticSub,
+   BinaryArithmeticMul,
+   BinaryArithmeticDiv,
+   BinaryArithmeticAnd,
+   BinaryArithmeticOr,
+   BinaryArithmeticXor,
+   BinaryArithmeticMin,
+   BinaryArithmeticMax,
+   BinaryLogicalShiftLeft,
+   BinaryLogicalShiftRight,
+   BinaryArithmeticShiftRight,
+   BinaryRotateLeft,
+   BinaryRotateRight,
+   NumBinaryArithmeticOps,
+   UnaryArithmeticAbs,
+   UnaryArithmeticSqrt,
+   LastOp,
+   NumUnaryArithmeticOps = LastOp - NumBinaryArithmeticOps + 1
+   };
+
+// TODO: Truncate the table
```

Comment:
Please create an issue describing details of future work and reference it here.  Otherwise, TODOs and their context get lost in the code.